node/node_newblock.go

```diff
@@ -84,6 +84,11 @@ func (node *Node) WaitForConsensusReadyV2(readySignal chan consensus.ProposalTyp
 					newBlock, err := node.ProposeNewBlock(newCommitSigsChan)
 
 					if err == nil {
+						if blk, ok := node.proposedBlock[newBlock.NumberU64()]; ok {
```

Comment:
what to do with this situation when the same number is proposed?  how to recover from it?

Reply:
We don't want to propose the same block, if it is tried, we just return here. If if a duplicate block is indeed proposed, the other checks about the block number will prevent issues downstream.